src/client.rs

```diff
@@ -0,0 +1,149 @@
+                let game = get_game(&client, game_url.as_str()).await;
+
+                println!("{}", game.board());
+            }
+            State::Win(winner) => {
+                println!("{winner} wins!");
+                return;
+            }
+            State::Tie => {
+                println!("Tie.");
+                return;
+            }
+        }
+    }
+}
+
+async fn get_or_create_game(client: &Client) -> Uuid {
+    println!("Create game (1) or join existing one (2)?");
+    let mut buffer = String::default();
+    stdin()
+        .read_line(&mut buffer)
+        .expect("failed to read the choice");
+    let choice: u32 = buffer
+        .trim()
+        .parse()
+        .expect(format!("expected a number, got {buffer}").as_str());
+
+    let game_id: Uuid = match choice {
+        1 => {
+            let string = client
+                .get(SERVER_URL)
+                .send()
+                .await
+                .expect("failed to request a new game")
+                .text()
+                .await
+                .expect("failed to get request body");
+            let id = Uuid::from_str(&string).expect("failed to parse UUID");
+            println!("game id: {id}");
+            id
+        }
+        2 => {
+            println!("Game id:");
+            let mut buffer = String::default();
+            stdin()
+                .read_line(&mut buffer)
+                .expect("failed to read the id");
+            let id: Uuid = buffer
+                .trim()
+                .parse()
+                .expect(format!("expected a UUID, got {buffer}").as_str());
+            id
+        }
+        other => panic!("expected 1 or 2, got {other}"),
+    };
+
+    game_id
+}
+
+fn get_player_role() -> Mark {
+    println!("Circle (1) or cross (2)?");
+    let mut buffer = String::default();
+    stdin()
+        .read_line(&mut buffer)
+        .expect("failed to read the role");
+    let choice: u32 = buffer
+        .trim()
+        .parse()
+        .expect(format!("expected a number, got {buffer}").as_str());
+    let player_role: Mark = match choice {
+        1 => Mark::Circle,
+        2 => Mark::Cross,
+        other => panic!("expected 1 or 2, got {other}"),
+    };
+    player_role
+}
+
+async fn get_game(client: &Client, game_url: &str) -> Game {
+    let game: Game = client
+        .get(game_url)
+        .send()
+        .await
+        .expect("failed to request game")
+        .json()
+        .await
+        .expect("failed to get request body");
+    game
+}
+
+fn get_move() -> (usize, usize) {
+    println!("Enter your move (row, column)");
+    let mut buffer = String::default();
+    stdin()
+        .read_line(&mut buffer)
+        .expect("failed to read the move");
+    let mut parts = buffer.split(' ');
+    let row: usize = parts
+        .next()
+        .expect(format!("expected two numbers, got {buffer}").as_str())
```

Comment:
I think client interaction code should be infallible, you should print explanation message and give additional input attempt. What will server do on client crash? Will another player of this game wait for next move forever?